stackslib/src/chainstate/burn/db/sortdb.rs

```diff
@@ -1931,19 +1938,40 @@ impl<'a> SortitionHandleConn<'a> {
     pub fn expects_stacker_signature(
         &self,
         consensus_hash: &ConsensusHash,
-        _stacker_signature: &MessageSignature,
+        stacker_signature: &SchnorrSignature,
+        message: &[u8],
     ) -> Result<bool, db_error> {
         // is this consensus hash in this fork?
         let Some(bhh) = SortitionDB::get_burnchain_header_hash_by_consensus(self, consensus_hash)?
         else {
             return Ok(false);
         };
-        let Some(_sortition_id) = self.get_sortition_id_for_bhh(&bhh)? else {
+        let Some(sortition_id) = self.get_sortition_id_for_bhh(&bhh)? else {
+            return Ok(false);
+        };
+        let Some(stackers_aggregate_public_key) = self.get_stackers_aggregate_public_key(&sortition_id)? else {
             return Ok(false);
         };
+        Ok(stacker_signature.verify(&stackers_aggregate_public_key, message))
+    }
 
-        // TODO: query set of stacker signers in order to get the aggregate public key
-        Ok(true)
+    /// Retrieve the stackers aggregate public key from the sortition DB
+    pub fn get_stackers_aggregate_public_key(
+        &self,
+        sortition_id: &SortitionId,
+    ) -> Result<Option<Point>, db_error> {
+        self.get_indexed(sortition_id, &db_keys::pox_stackers_aggregate_public_key())
+            .map(|x| {
+                x.map(|x| {
+                    let x = hex_bytes(&x).expect("CORRUPTION: bad stackers aggregate public key. Invalid hex string.");
+                    let mut x_bytes: [u8; 33] = [0; 33];
+                    if x_bytes.len() != x.len() {
+                        panic!("CORRUPTION: bad stackers aggregate public key. Expected {} bytes, got {} bytes", x_bytes.len(), x.len());
+                    }
+                    x_bytes.copy_from_slice(&x[0..33]);
+                    Point::try_from(&Compressed::from(x_bytes)).expect("CORRUPTION: bad stackers aggregate public key")
+                })
+            })
```

Comment:
Not a huge issue, but I think in a case with nested `map()` invocations, it's probably more clear to flatten the nesting use the try syntax `?` instead:

```suggestion
        let Some(pk_string) = self.get_indexed(sortition_id, &db_keys::pox_stackers_aggregate_public_key())?
        else {
              return Ok(None)
        };
        let pk_byte_vec = hex_bytes(&pk_string).expect("CORRUPTION: bad stackers aggregate public key. Invalid hex string.");
        let mut pk_bytes: [u8; 33] = [0; 33];
        if pk_bytes.len() != pk_byte_vec.len() {
             panic!("CORRUPTION: bad stackers aggregate public key. Expected {} bytes, got {} bytes", x_bytes.len(), x.len());
        }
        pk_bytes.copy_from_slice(&pk_byte_vec[0..33]);
        Ok(Point::try_from(&Compressed::from(pk_bytes)).expect("CORRUPTION: bad stackers aggregate public key"))
```

Reply:
Yeah, it's generally a good idea to avoid excessive indentations